device/transport/mqtt/test/_mqtt_receiver_test.js

```diff
@@ -329,5 +330,24 @@ describe('Mqtt as MqttReceiver', function () {
         });
       });
     });
+
+    /*Tests_SRS_NODE_DEVICE_MQTT_18_057: [ If there is a listener for the `inputMessage` event, a `inputMessage` event shall be emitted for each message received. ]*/
+    /*Tests_SRS_NODE_DEVICE_MQTT_18_058: [ When an `inputMessage` event is received, Mqtt shall extract the inputName from the topic according to the following convention: 'devices/<deviceId>/modules/<moduleId>/inputs/<inputName>' ]*/
+    /*Tests_SRS_NODE_DEVICE_MQTT_18_056: [ When an `inputMessage` event is emitted, the first parameter shall be the inputName and the second parameter shall be of type `Message`. ]*/
+    describe('#inputMessage', function() {
+        it('emits an inputMessage event with a Message object when there is a listener', function (done) {
+          fakeConfig.moduleId = 'moduleId';
+          var receiver = new Mqtt(fakeAuthenticationProvider, fakeMqttBase);
+          receiver.connect(function () {
+            receiver.on('inputMessage', function (inputName, msg) {
+              assert.strictEqual(inputName, 'fakeInputName')
+              assert.instanceOf(msg, Message);
```

Comment:
>assert.instanceOf(msg, Message); [](start = 14, length = 32)

is that working? you're not passing a message in the emit?

Reply:
Yep, this was copy/paste from the corresponding message test

---
In reply to: [171429827](https://github.com/Azure/azure-iot-sdk-node/pull/237#discussion_r171429827) [](ancestors = 171429827)